File: Axiom/Internal/Intermediate/ImExpressionSequence.cs
﻿using System.IO;

namespace Axiom.Internal.Intermediate
{
    internal class ImExpressionSequence : ImExpression
    {
        public ImExpressionSequence(ImStatement statement, ImExpression expression)
        {
            Statement = statement;
            Expression = expression;
        }

        public override void Dump(TextWriter tw, int ident)
        {
            Report.Dump(tw, ident, "ExpressionSequence");
            Statement.Dump(tw, ident + 1);
            Expression.Dump(tw, ident + 1);
        }

        public override ImExpressionSequence Linear()
        {
            var linearStmt = Statement.Linear();
            var linearExpr = Expression.Linear();

            linearStmt.Statements.AddAll(((ImSequence)linearExpr.Statement).Statements);
            linearExpr.Statement = linearStmt;

            return linearExpr;
        }

        public ImStatement Statement { get; set; }
        public ImExpression Expression { get; set; }
    }
}

File: src/main/java/com/moybl/axiom/im/ImExpressionSequence.java
package com.moybl.axiom.im;

public class ImExpressionSequence implements ImExpression{

	private ImStatement statement;
	private ImExpression expression;

	public ImExpressionSequence(ImStatement statement, ImExpression expression) {
		this.statement = statement;
		this.expression = expression;
	}

	@Override
	public void accept(ImVisitor visitor) {
		visitor.visit(this);
	}

	public ImStatement getStatement() {
		return statement;
	}

	public ImExpression getExpression() {
		return expression;
	}

}
